code/game/atoms.dm

```diff
@@ -163,6 +163,20 @@
 
 	return FALSE
 
+///General check to see if an object is on an away mission, returns FALSE if it can't find a turf or if not on an away mission z-level.
+/atom/proc/onAwayMission()
+	var/turf/T = get_turf(src)
+	if(!T)
+		return FALSE
+
+	if(!is_away_level(T.z))
+		return FALSE
+
+	if(istype(T.loc, /area/awaymission))

```

Comment:
This check seems pointless with the z level check. Also I'm not sure if every away mission only has away mission areas. I wouldn't be too surprised if at some point even someone gets a shuttle into an away mission.